packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/helpers/golden_helper.dart

```diff
@@ -17,69 +23,56 @@ const localizations = [
+  Device.tabletLandscape,
+];
+
+/// Convenience method that builds a [ScenarioBuilder] with a scenario rendered
+/// on specified devices by default
+ScenarioBuilder buildScenario({
   required Widget widget,
-  required Scenario scenario,
-}) {
-  final deviceBuilder = LabeledDeviceBuilder(label: scenario.name)
-    ..overrideDevicesForAllScenarios(
-      devices: [
-        Device.phone,
-        Device.iphone11,
-        Device.tabletPortrait,
-        Device.tabletLandscape,
-      ],
-    )
-    ..addScenario(
+  required String scenario,
+  List<Device> devices = _defaultDevices,
+  int? columns,
+}) =>
+    ScenarioBuilder(
+      name: scenario,
       widget: widget,
-      name: scenario.name,
-      onCreate: scenario.onCreate,
+      devices: devices,
+      columns: columns,
     );
-  return deviceBuilder;
-}
 
-/// executes golden tests for each [LabeledDeviceBuilder] in every [theme]
-///
-/// [deviceBuilders] - list of [LabeledDeviceBuilder] to be pumped
-///
-/// [pumpFunction] (optional) - function for executing custom pumping
-/// behavior instead of [pumpDeviceBuilderWithLocalizationsAndTheme]
+/// Runs golden tests for a list of scenarios in both light and dark mode
 void runGoldenTests(
-    List<LabeledDeviceBuilder> deviceBuilders, {
-      Future<void> Function(WidgetTester, DeviceBuilder, Themes? theme)?
-      pumpFunction,
-      CustomPump? matcherCustomPump,
+    List<ScenarioBuilder> buildScenarios, {
+      WidgetWithThemePump? customWrapAndPump,
+      WidgetTesterCallback? act,
     }) {
-  for (final db in deviceBuilders) {
-    //test each DeviceBuilder in both light mode and dark mode
+  for (final scenario in buildScenarios) {
     for (final theme in Themes.values) {
       final themeName = theme.name;
-      final directory = '${themeName}_theme';
-
-      testGoldens('$db - $themeName', (tester) async {
-        pumpFunction != null
-            ? await pumpFunction.call(tester, db, theme)
-            : await pumpDeviceBuilderWithLocalizationsAndTheme(
-          tester,
-          db,
-          theme: theme,
-        );
+      final scenarioName = scenario.name;
 
-        await screenMatchesGolden(
-          tester,
-          '$directory/$db',
-          //defaults to pumpAndSettle, causing problems when testing animations
-          customPump: matcherCustomPump ??
-              (db.label.contains('loading')
-                  ? (tester) => tester.pump(const Duration(microseconds: 300))
-                  : null),
-        );
-      });
+      goldenTest(
+        '$scenarioName - $themeName',
+        fileName: '${scenarioName}_$themeName',
+        builder: () => scenario,
+        pumpWidget: (tester, widget) =>
+        customWrapAndPump?.call(tester, widget, theme) ??
+            pumpDeviceBuilderWithLocalizationsAndTheme(
+              tester,
+              widget,
+              theme: theme,
+            ),
+        pumpBeforeTest: scenarioName.contains('loading')
```

Comment:
I believe we need to think for a better way of handling animations

Reply:
Sure thing; I am open to new suggestions. As a note here I'd like to add that this was taken from the previous implementation with the `golden_toolkit` package, so it may be an older approach.

Reply:
I have no suggestions. Please think for a way of handling animations and then we can discuss it

Reply:
> Also you can test how the animations behave in the alchemist package. This workaround might be goolden_toolkit-specific only.

I'll check and see how it affects the tests without it.

---

As for options, I have some ideas:
1. Add an optional argument to the `runGoldenTests` function to serve as a custom pump.

```dart
void runGoldenTests(
    List<ScenarioBuilder> buildScenarios, {
      WidgetWithThemePump? customWrapAndPump,
      WidgetTesterCallback? act,
      Future<void> Function(WidgetTester)? customPumpBeforeTest,  // <---- something like this
    })
```

The good and the bad side to this would be that we define one custom pump function for all the build scenarios.

2. Add an optional custom pump argument to the `buildScenario` function

```dart
ScenarioBuilder buildScenario({
  required Widget widget,
  required String scenario,
  List<Device> devices = _defaultDevices,
  int? columns,
  Future<void> Function(WidgetTester)? customPumpBeforeTest,  // <---- something like this
}) 
```

The good side to this approach is that we can have different scenarios and adjust them accordingly.
The downside to this approach is that if we have a number of scenarios, we'd have to specify the function for each of them.

3. Combine approaches from `1.` and `2.` but prioritise custom pump methods from individual scenarios.

This would fix both problems from second point:
- We apply the custom pump (if provided) from the `runGoldenTests` function to all scenarios.
- If a scenario defines its own custom pump, then use that one.
- If there aren't any custom pumps provided, use the default one.

Reply:
How about implementing a global function, which can be passed as method reference to the `buildScenario` method ? Something like this 
```
buildScenario(
    widget: ...,
    scenario: ...,
    customPumpBeforeTest: animationCustomPump
)
```

Reply:
Implemented!